packages/server-core/src/media/storageprovider/s3.storage.ts

```diff
@@ -200,19 +215,21 @@ export class S3Provider implements StorageProviderInterface {
     if (data.Metadata) (args as StorageObjectInterface).Metadata = data.Metadata
 
     if (data.Body instanceof PassThrough) {
-      return new Promise<AWS.S3.ManagedUpload.SendData>((resolve, reject) => {
-        const upload = this.provider.upload(args)
+      try {
+        const upload = new Upload(args as unknown as Options)
+
         upload.on('httpUploadProgress', (progress) => {
           console.log(progress)
           // if (params.onProgress) params.onProgress(progress.loaded, progress.total)
         })
-        upload.send((err, data) => {
-          if (err) reject(err)
-          else resolve(data)
-        })
-      })
+        await upload.done()
```

Comment:
Need to return this; if you look at what it's replacing, we were resolving the container Promise with the `data` value that resulted from upload.send.